test/game/logic/conveyor_utilityTests.cpp

```diff
@@ -0,0 +1,899 @@
+
+        {
+            TestSetupConveyor(worldData_, {0, 0}, Orientation::right, transBelt_);
+
+            worldData_.LogicRegister(kLogicGroup_, {0, 0}, TileLayer::entity);
+        }
+
+        ConveyorDestroy(worldData_, {0, 0}, kLogicGroup_);
+
+        auto* con_data = worldData_.GetTile({0, 0})->GetLayer(TileLayer::entity).GetUniqueData<proto::ConveyorData>();
+        ASSERT_NE(con_data, nullptr);
+        EXPECT_EQ(con_data->structure, nullptr);
+
+        EXPECT_EQ(worldData_.GetChunkW({0, 0})->GetLogicGroup(kLogicGroup_).size(), 0);
+    }
+
+    ///
+    /// Removing beginning of grouped conveyor segment with bending termination
+    /// Creates new segment with what was the second tile now the head
+    TEST_F(ConveyorUtilityTest, DestroyHeadBendingTermination) {
+        //
+        // > > v
+        //
+
+        {
+            auto& head_con_data       = TestSetupConveyor(worldData_, {1, 0}, Orientation::right, transBelt_);
+            head_con_data.structIndex = 1;
+            head_con_data.structure->terminationType = ConveyorStruct::TerminationType::bend_right;
+
+            TestSetupConveyor(worldData_, {0, 0}, transBelt_, head_con_data.structure);
+
+            TestSetupConveyor(worldData_, {2, 0}, Orientation::down, transBelt_);
+
+
+            worldData_.LogicRegister(kLogicGroup_, {1, 0}, TileLayer::entity);
+            worldData_.LogicRegister(kLogicGroup_, {2, 0}, TileLayer::entity);
+        }
+
+        ConveyorDestroy(worldData_, {1, 0}, kLogicGroup_);
+
+
+        auto* behind_con_data = GetConData(worldData_, {0, 0});
+        ASSERT_NE(behind_con_data, nullptr);
+
+        EXPECT_EQ(behind_con_data->structure->length, 1);
+        // Unaffected since this tile will be removed
+        EXPECT_EQ(behind_con_data->structure->terminationType, ConveyorStruct::TerminationType::bend_right);
+
+        EXPECT_EQ(worldData_.GetChunkW({0, 0})->GetLogicGroup(kLogicGroup_).size(), 1);
+    }
+
+    ///
+    /// Removing middle of grouped conveyor segment
+    /// Create new segment behind, shorten segment ahead
+    TEST_F(ConveyorUtilityTest, DestroyMiddle) {
+        //
+        // > /> > >
+        //
+
+        {
+            auto& head_con_data                 = TestSetupConveyor(worldData_, {3, 0}, Orientation::right, transBelt_);
+            head_con_data.structure->headOffset = 30; // Will use this to set structure behind itemOffset
+            head_con_data.structure->length     = 4;  // Will use this to set structure behind length
+
+            TestSetupConveyor(worldData_, {2, 0}, transBelt_, head_con_data.structure);
+
+            auto& con_data_3       = TestSetupConveyor(worldData_, {1, 0}, transBelt_, head_con_data.structure);
+            con_data_3.structIndex = 2; // Will use this to set structure ahead length
+
+            auto& con_data_4       = TestSetupConveyor(worldData_, {0, 0}, transBelt_, head_con_data.structure);
+            con_data_4.structIndex = 100; // Should be changed
```

Comment:
Use values matching actual conveyors,

make the length 5 and remove the middle to ensure the headOffset is actually changed